packages/legend-shared/src/CommonUtils.ts

```diff
@@ -224,6 +224,14 @@ export const shallowStringify = (object: unknown): string =>
       : val,
   );
 
+export const validateValidString = (input: string): boolean => {
+  if (input.match(/^[a-z][\w\d]*$/)) {
```

Comment:
I thought `\w` includes `\d`? - Also I think we diverge here, Pure/Engine allows `$` after the first char <---- let's make it optional check for us